llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2493,19 +2499,20 @@ class VPReplicateRecipe : public VPRecipeWithIRFlags {
 public:
   template <typename IterT>
   VPReplicateRecipe(Instruction *I, iterator_range<IterT> Operands,
-                    bool IsUniform, VPValue *Mask = nullptr)
+                    bool IsUniform, VPValue *Mask = nullptr,
+                    ArrayRef<std::pair<unsigned, MDNode *>> Metadata = {})
```

Comment:
Note: if a default/empty Metadata is passed, should getMetadataToPropagate(I) be used?

Reply:
Probably best done separately, as for now the existing metadata is ignored in VPIRMetadata, except  new noalias.